lib/screens/create_region_screens/choose_region_name/choose_region_name.dart

```diff
@@ -1,41 +1,67 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:seeds/components/flat_button_long.dart';
+import 'package:seeds/components/full_page_error_indicator.dart';
+import 'package:seeds/components/full_page_loading_indicator.dart';
 import 'package:seeds/components/text_form_field_custom.dart';
 import 'package:seeds/design/app_theme.dart';
+import 'package:seeds/domain-shared/page_state.dart';
 import 'package:seeds/domain-shared/ui_constants.dart';
+import 'package:seeds/screens/create_region_screens/viewmodels/create_region_bloc.dart';
 import 'package:seeds/utils/build_context_extension.dart';
 
 class ChooseRegionName extends StatelessWidget {
   const ChooseRegionName({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-        appBar: AppBar(
-          title: Text(context.loc.createRegionSelectRegionAppBarTitle),
-        ),
-        body: SafeArea(
-            minimum: const EdgeInsets.all(horizontalEdgePadding),
-            child: Stack(
-              children: [
-                Column(
-                  children: [
-                    const SizedBox(height: 10),
-                    TextFormFieldCustom(
-                      autofocus: true,
-                      labelText: context.loc.createRegionChooseRegionNameInputFormTitle,
-                    ),
-                    const SizedBox(height: 20),
-                    Text(context.loc.createRegionChooseRegionNameDescription,
-                        style: Theme.of(context).textTheme.subtitle2OpacityEmphasis),
-                  ],
-                ),
-                Align(
-                  alignment: Alignment.bottomCenter,
-                  child: FlatButtonLong(
-                      title: "${context.loc.createRegionSelectRegionButtonTitle} (2/5)", onPressed: () {}),
+    return BlocBuilder<CreateRegionBloc, CreateRegionState>(builder: (context, state) {
+      Future<bool> _navigateBack() {
+        BlocProvider.of<CreateRegionBloc>(context).add(const OnBackPressed());
+        return Future.value(false);
+      }
```

Comment:
Same and duplicated code can we handle this bloc event with a listener or just without a function for both willpop and back button?

Reply:
removed function